src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -748,6 +751,53 @@ public ResponseEntity<List<UserDTO>> searchStudentsInCourse(@PathVariable Long c
         return new ResponseEntity<>(page.getContent(), headers, HttpStatus.OK);
     }
 
+    /**
+     * GET /courses/:courseId/users/search : Search all users by login or name that belong to the specified groups of the course
+     *
+     * @param courseId    the id of the course
+     * @param loginOrName the login or name by which to search users
+     * @param roles       the roles which should be searched in
+     * @return the ResponseEntity with status 200 (OK) and with body all users
+     */
+    @GetMapping("/courses/{courseId}/users/search")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<List<UserPublicInfoDTO>> searchUsersInCourse(@PathVariable Long courseId, @RequestParam("loginOrName") String loginOrName,
+            @RequestParam("roles") List<String> roles) {
+        // ToDo: Discuss what information should be hidden as this is a public endpoint
```

Comment:
Open todo.

Reply:
> Looks good, joined the test session. Little visibility changes like color and border changes for messaging part would be nice in order to help visually declare where the messaging window begins (was gone over in the test session).

I adapted the color and added a border to the messaging window. As requested in the testing session I also added the ability of channel moderators to edit and delete messages in a channel.

Reply:
Removed the todo in 7774f6c5d3d4504aa56ec1626100926ac97057f7

Already discussed. We are sending the minimal information which just includes login and the name. With less we can not differentiate between users with the same name in the chat.